src/View/Form/ContextInterface.php

```diff
@@ -44,6 +44,15 @@ public function isPrimaryKey($field);
     public function isCreate();
 
     /**
+     * Sets the type of the form the context is created from.
+     *
+     * @param string $requestType The request type of the form the
+     *   context is created from.
+     * @return void
+     */
+    public function setRequestType($requestType);
```

Comment:
Yes. Stick to the problems we have today. Having multiple modes means more options, more documentation and more tests required. We can always add more complexity later, but removing complexity/configuration/features is really hard.